app/src/components/common/layout/header/index.tsx

```diff
@@ -99,46 +137,155 @@ const ContentsLeft = styled.div`
 const ContentsRight = styled.div`
   align-items: center;
   display: flex;
+  flex-wrap: wrap;
   margin: auto 0 auto auto;
 
   @media (min-width: ${props => props.theme.themeBreakPoints.md}) {
     margin: auto 0 0 auto;
+    flex-wrap: unset;
   }
 `
 
 const HeaderDropdown = styled(Dropdown)`
-  ${ButtonCSS}
+  ${ButtonCSS};
 `
 
 const CloseIconWrapper = styled.div`
   margin-right: 12px;
 `
+const ClaimWrapper = styled.div`
+  display: flex;
+  width: 100%;
+  justify-content: space-between;
+`
+const ClaimAmount = styled.div`
+  color: ${props => props.theme.colors.textColorDark};
+  font-size: ${props => props.theme.fonts.defaultSize};
+`
+const GoldDot = styled.div`
+  height: 6px;
+  width: 6px;
+  background-color: ${props => props.theme.colors.gold};
+  border-radius: 50%;
+  display: inline-block;
+  margin-right: 8px;
+`
+const ClaimText = styled.div`
+  display: flex;
+  align-items: center;
+`
+
+interface ExtendsHistory extends RouteComponentProps {
+  setClaim: React.Dispatch<React.SetStateAction<boolean>>
+}
+
+function useOutsideAlerter(ref: any, setState: any) {
+  useEffect(() => {
+    /**
+     * Alert if clicked on outside of element
+     */
+    function handleClickOutside(event: any) {
+      if (ref.current && !ref.current.contains(event.target)) {
+        setState(false)
+      }
+    }
+
+    // Bind the event listener
+    document.addEventListener('mousedown', handleClickOutside)
+    return () => {
+      // Unbind the event listener on clean up
+      document.removeEventListener('mousedown', handleClickOutside)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [ref])
+}
```

Comment:
Might be good to move this into the hooks directory so it can be reused elsewhere

Reply:
yeah i have some error because of passing ref as a prop ....spent bunch of time on it since you commented...i got it to work but it clogs console with errors...will handle it tommorow!